Examples/file-manager.lisp

```diff
@@ -0,0 +1,260 @@
+  (make-instance 'directory* :name name :children children))
+
+(defmethod adopt ((parent directory*) (child file))
+  (setf (directory* child) parent)
+  (push child (children parent)))
+
+(defmethod disown ((parent directory*) (child file))
+  (setf (directory* child) nil)
+  (alexandria:removef (children parent) child))
+
+(defclass root (directory*) ())
+
+(defun make-root (&rest children)
+  (make-instance 'root :name "<root>" :children children))
+
+;;; Presentation methods
+
+(defclass file-manager-view (view) ())
+
+(define-presentation-method present ((object file)
+                                     (type   file)
+                                     (stream extended-output-stream)
+                                     (view   file-manager-view)
+                                     &key)
+  (surrounding-output-with-border (stream :shape :rectangle :background +yellow+)
+    (princ (name object) stream)))
+
+(defun print-directory-contents (items stream)
+  (formatting-item-list (stream)
+    (map nil (lambda (child)
+               (formatting-cell (stream)
+                 (present child (presentation-type-of child)
+                          :stream stream :single-box t)))
+         (sort (copy-list items) #'string< :key #'name))))
+
+(define-presentation-method present ((object directory*)
+                                     (type   directory*)
+                                     (stream extended-output-stream)
+                                     (view   file-manager-view)
+                                     &key)
+  (surrounding-output-with-border (stream :shape :rectangle :background +beige+)
+    (princ (name object) stream)
+    (with-translation (stream (+ (stream-cursor-position stream) 8) 0)
+      (draw-rectangle* stream 0 0 16 16))
+    (terpri stream)
+    (print-directory-contents (children object) stream)))
+
+(define-presentation-method present ((object root)
+                                     (type   root)
+                                     (stream extended-output-stream)
+                                     (view   file-manager-view)
+                                     &key)
+  (with-bounding-rectangle* (x1 y1 x2 y2) (sheet-region stream)
+    (draw-rectangle* stream (+ x1 4) (+ y1 4) (- x2 4) (- y2 4) :filled nil))
+  (stream-increment-cursor-position stream 8 8)
+  (print-directory-contents (children object) stream))
+
+;;; Application
+
+(define-application-frame file-manager ()
+  ((%root :initarg  :root
+          :reader   root
+          :initform (make-root
+                     (make-directory "Trash")
+                     (make-directory "Directory 1")
+                     (make-directory "Directory 2"
+                                     (make-file "baz")
+                                     (make-file "fez"))
+                     (make-file "whoop"))))
+  (:panes
+   (explananation :label :label "Drag files and directories. Modifier keys change move, copy, etc.")
```

Comment:
more elaborate explanation (which modifier does what). Also "Modifier keys change > drop action to < …"

Reply:
Done (thanks to multi-line labels).